api/v1alpha1/amaltheasession_children.go

```diff
@@ -101,20 +102,53 @@ func (cr *AmaltheaSession) StatefulSet() appsv1.StatefulSet {
 	}
 
 	securityContext := &v1.SecurityContext{
-		RunAsNonRoot: &[]bool{true}[0],
-		RunAsUser:    &[]int64{session.RunAsUser}[0],
-		RunAsGroup:   &[]int64{session.RunAsGroup}[0],
+		RunAsNonRoot: ptr.To(true),
+		RunAsUser:    ptr.To(session.RunAsUser),
+		RunAsGroup:   ptr.To(session.RunAsGroup),
 	}
 
 	if session.RunAsUser == 0 {
-		securityContext.RunAsNonRoot = &[]bool{false}[0]
+		securityContext.RunAsNonRoot = ptr.To(false)
 	}
 
 	sessionContainer.SecurityContext = securityContext
 
 	containers := []v1.Container{sessionContainer}
 	containers = append(containers, cr.Spec.ExtraContainers...)
 
+	if cr.Spec.Authentication != nil && cr.Spec.Authentication.Enabled {
+		auth := cr.Spec.Authentication
+
+		if auth.Type == Oidc {
+			volumes = append(volumes, v1.Volume{
+				Name: "proxy-configuration-secret",
+				VolumeSource: v1.VolumeSource{
+					Secret: &v1.SecretVolumeSource{
+						SecretName: auth.SecretRef.Name,
+						Optional:   ptr.To(false),
+					},
+				},
+			})
+
+			authContainer := v1.Container{
+				Image: "bitnami/oauth2-proxy:7.4.0",
+				Name:  "oauth2-proxy",
+				SecurityContext: &v1.SecurityContext{
+					AllowPrivilegeEscalation: ptr.To(false),
+					RunAsNonRoot:             ptr.To(true),
+				},
+				Args: []string{"--config=/etc/oauth2-proxy/" + auth.SecretRef.Key},
+				VolumeMounts: []v1.VolumeMount{
+					{
+						Name:      "proxy-configuration-secret",
+						MountPath: "/etc/oauth2-proxy",
+					},
+				},
+			}
+			containers = append(containers, authContainer)
+		}
+	}
```

Comment:
Maybe for the sake of concisenenss

```suggestion
	if auth := cr.Spec.Authentication; auth != nil && auth.Enabled && auth.Type == Oidc {

		volumes = append(volumes, v1.Volume{
			Name: "proxy-configuration-secret",
			VolumeSource: v1.VolumeSource{
				Secret: &v1.SecretVolumeSource{
					SecretName: auth.SecretRef.Name,
					Optional:   ptr.To(false),
				},
			},
		})

		authContainer := v1.Container{
			Image: "bitnami/oauth2-proxy:7.4.0",
			Name:  "oauth2-proxy",
			SecurityContext: &v1.SecurityContext{
				AllowPrivilegeEscalation: ptr.To(false),
				RunAsNonRoot:             ptr.To(true),
			},
			Args: []string{"--config=/etc/oauth2-proxy/" + auth.SecretRef.Key},
			VolumeMounts: []v1.VolumeMount{
				{
					Name:      "proxy-configuration-secret",
					MountPath: "/etc/oauth2-proxy",
				},
			},
		}
		containers = append(containers, authContainer)
	}
```